test/test_reductions.py

```diff
@@ -884,6 +884,14 @@ def test_mean_int_with_optdtype(self, device):
         a_float = a.to(torch.float32)
         self.assertEqual(a_float.mean(), a.mean(dtype=torch.float32))
 
+    @onlyCPU
+    def test_mean_out_float16_is_alias_of_return(self, device):
+        a = torch.randn((3, 1, 4), dtype=torch.float16, device=device)
+        out = torch.empty((1, 1, 4), dtype=torch.float16, device=device)
+
+        return_out = torch.mean(a, dim=0, keepdim=True, out=out)
+        self.assertTrue(torch._C._is_alias_of(out, return_out))
```

Comment:
Thanks for adding the test, but maybe also a small check for correctness by comparing with the out-of-place mean op would be good?